shared/js/font_size_utils.js

```diff
@@ -208,12 +216,22 @@
       var resultWidth;
       var overflowCount = -1;
 
+      if (string.length > this._overflowCountCache) {
+        substring = string.substr(0, string.length - this._overflowCountCache);
+        resultWidth = this.getFontWidth(substring, fontSize, fontFamily);
```

Comment:
Now that repeat this logic twice, perhaps we should abstract it into a function:
```
resultWidth = this.getSubstringFontWidth(string, endIndex, fontSize, fontFamily);